### DISCUSSION TITLE
ZuluSCSI RP2040 Speaker Options

### DISCUSSION THREAD

**Discussion by SezaroSteve** [ZuluSCSI/ZuluSCSI-firmware#132] [General]
Hello,

I noticed there is an optional speaker location on the board of the regular RP2040. I can't find any info about it anywhere. Is speaker support implemented yet? If so, what speaker (and other necessary components) could I use?

Kind regards,
Steve

**Comment by PetteriAimonen** [comment]
The speaker is connected to same control line as the status LEDs, each with their own resistor.
You can use a small electromagnetic or piezo speaker, and it will make a bit of rattle whenever the disk is accessed.
Loudness will depend on speaker model and the series resistor value (62 ohm by default).

The KiCAD footprint has a hole distance of 7.6 mm, which is apparently less common than 6.5 mm.
In any case, difference is small enough that either type can be soldered on, or you can connect wires to an external speaker.

I have tried it with LD-BZEN-1212, which produces volume that is comparable to mechanical harddrives.

**Reply by SezaroSteve** [reply]
1212 means 12mm Diameter and 12V I guess. Why did you choose that particular model? (LD-BZEN-1212)

I found a electromagnetic unit, the "BeStar Acoustic BMT-1212UX Magnetic Transducer" which has very similar specs as the Loudity 1212 you mentioned.  The LD-BZEN-1212 has a resonant frequency of 2400 Hz. The BMT-1212UX is 2300 Hz (+- 400Hz) unit. The pins on the BMT-1212UX are 7.6mm apart so a perfect fit for the regular (non mini) RP2040 board. Would the BMT unit work?

Another 12V 12mm unit I found is the Loudity LD-BZEG-1212 which is a Magnetic Transducer with built-in generator (the LD-BZEN-1212 is without). It too has a frequency of 2300 Hz (+- 400Hz) and a 7.6mm terminal pitch. There is a video on Youtube of what the LD-BZEG-1212 sounds like and it's just a high pitched beeeeeeeeeeeb. I guess that's because they are probably giving it constant power instead of pulses. But I don't think it will "rattle" (as you describe it) or gently click like a gotek buzzer (that sound would not be a bad choice as sometimes, a gotek with buzzer does sound like an old HDD).

With or without built-in generator. I have no idea what to choose or why. I only half understand what i'm writing by the way, so if i'm talking gibberish, be gentle with me :-)

**Reply by PetteriAimonen** [reply]
BMT-1212UX and  LD-BZEG-1212 have internal sound generator, so they will make a beep instead of a click.
The ones without a generator are just plain speakers, so they'll click when the status LED comes on and off.

The voltage rating of the non-generator ones affects only sound volume - ZuluSCSI drives the speaker at 3V, so a 12V model will not be as loud as a 3V model would be. The LD-BZEN-1212 I just happened to have from another project.

**Comment by aperezbios** [comment]
@SezaroSteve the TDK PS1240P02BT piezo buzzer is what we recommend. 

It can be purchased from [Mouser](https://www.mouser.com/ProductDetail/TDK/PS1240P02BT?qs=d7g9p1yFhWaZXSY9MjKMkw%3D%3D) and [Digi-Key 
](https://www.digikey.com/en/products/detail/tdk-corporation/PS1240P02BT/935924) at a cost of around $0.50 to $0.60 each.

[Adafruit also stocks them](https://www.adafruit.com/product/160) for $1.50 each.

As others have noted, the lead spacing on this TDK piezo buzzer isn't the same as the lead spacing on the Rev 2022d RP2040 PCB, which doesn't prevent use of the buzzer, but it does mean you'll need to bend the leads before soldering it down, and that it won't sit 100% flush with the board. This will be fixed in a future revision of the PCB design.

**Reply by SezaroSteve** [reply]
Cool thanks. Will give it a try.
Question:  can I just solder it in, or does it require some component inline, like a resistor? (to reduce the volume maybe). That TDK is a 3V unit and then produces 70 dBa. I don't know what voltage is on the speaker lead of the RP2040 board.

**Reply by PetteriAimonen** [reply]
There is resistor R306 already on board. Default value is 62 ohms, which gives a volume level that is similar to mechanical drives.

**Reply by croissantking** [reply]
Any plans for a piezo speaker option on the PowerBook version of the zuluscsi rp2040? I would love to see this.

**Comment by beezbo** [comment]
Is there an audio / video demo of a piezo installed on the board? I am curious what it sounds like before trying to buy and install one.

**Reply by croissantking** [reply]
Here is a demo with the TDK part installed:

https://imgur.com/a/Hehla1Y

**Comment by jbehrends** [comment]
> Is there an audio / video demo of a piezo installed on the board? I am curious what it sounds like before trying to buy and install one.

Below is a closeup pic of the recommended TDK piezo buzzer installed, in addition J306 needs to be populated with a jumper so you can enable or disable the speaker.  Total cost in parts is about $1... And if you don't like the sound, pull the jumper!

Sorry, I don't have a video... But I'd say the sound this speaker creates is very similar to that of a laptop harddrive you'd find in a standard 2000's Laptop.   Mine is installed in an IBM PS/2 and if I hadn't installed this myself, I wouldn't have any clue that wasn't a real drive clicking away inside...  Just my 2 cents.

Here is a picture of the piezo buzzer and jumper header installed in J306: https://imgur.com/6eTKYnl

Good luck!

**Reply by SezaroSteve** [reply]
Listening to the demo, if i'm very honest, it does not sound like a harddisk at all. Not even close.
It doesn't "click away". It sounds more like the crackle/fizzle of something short-circuiting.

**Reply by croissantking** [reply]
It’s a little different to how I imagined as well. To me it sounds more like electronic noise (like capacitor whine) than a mechanical sound. Overall, I am happy with it because it breaks the dead silence on HDD activity that I find awkward.

I’m interested in experimenting with different buzzers/clickers, is there one that would sound more mechanical than the TDK?

**Reply by SezaroSteve** [reply]
I use the "Beleths Drum" in all my HDD equipped Amiga's (all ZuluSCSI). It's a fantastic solution. Sounds like those old Quantum LPS SCSI Drives from yesteryear. It even makes a "start-up sound" akin to the Quantum calibrating the heads during spin-up.

One can change how the Beleths Drum device sounds, by mounting the tiny circuit-board on a piece of metal to make it resonate more (isolate the unit from the metal to avoid short-circuiting of course). That way one can create a metallic clunking sound like those really old HDD's made. Make it sound like them full-height 5.25 from way-back-when.
Or one can soften it up, muffle it by mounting it on a piece of foam. For example: in the chassis/case of my Amiga 4000T, it is suspended in mid air inside the case, otherwise it's too loud. The 4000T's huge case acts as a echo-chamber whereas my 3000D has it foam-mounted against the backing-plate at the rear.
The device is powered by a standard Floppy power-connector. The signal from the ZuluScsi comes from 2 wires (+ and ground) off the ZuluSCSI's circuit-board LED-Header (I soldered a 2-pin jumper lead on to the board).

https://tinkerdifferent.com/threads/smc-presents-beleths-drum-bluescsi-hdd-sound-emulator.1124/

I got my units directly from Japan:  https://www.infinityproducts.co.jp/product-page/smc-beleth-s-drum-hd-sound-simulator


The problem with the buzzer on the Zulu's board is that the frequency with which it makes a sound, every time the SCSI bus is accessed), is way to high. That is why it sounds like the crackle/fizzle of something short-circuiting.
The Beleths drum slows the audio output way down, more akin to the speed with which RW-heads moved inside old HDD's. So instead of a "bzzzzzzzzzz" sound of a buzzer, it's more rattatattattattattatattatta of the Disks of old. It sounds way more realistic.

**Reply by croissantking** [reply]
Looks very interesting and I’ll probably be up for getting one at some point. But $55 shipped from Japan, vs $1-2 for the TDK part is not really a fair comparison.

**Reply by SezaroSteve** [reply]
Due to the shipping-costs, I purchased a bunch of them in one go. These devices are very small and light so when packed together, it's doable. I found it totally worth it.

About the TDK vs. the Beleths Drum: sorry but that is like comparing a super cheap car versus something decent. Not really a fair comparison.
I have spent a lot of money on these Amiga's, restoring them and all (it's not a cheap hobby by any stretch of the imagination) and then cheaping out on something that sounds pretty bad? (a 2$ buzzer). That does not compute for me.
I either do something right, or I don't do it at all. But to each his own.

**Reply by croissantking** [reply]
How do you mount them? Is there a tidy way when connected to the Zulu?

**Reply by SezaroSteve** [reply]
In the 4000T, I hang it off the Floppy-power-wires. It dangles in mid-air.
The connector and the 4 wires are more than strong enough to hold it (the device is very light). The 2 wires going to the ZuluSCSI is routed nicely through the case. Those wires do not contribute to holding the device in place.

In the 3000D, I cut a 4mm think piece of foam-material, the same size as the Beleth's circuit-board, and used a thin tie-wrap, wrapped around the foam and the board to hold them together. The back of the foam is held to the 3000's case (at the back, where the HDD used to be) by means of double-sided sticky-tape.

In my Desktoppified A1200, I used one of the two screw-holes on the perimeter of the circuit-board to screw it to a mounting-bracket.
Sorry it's all a bit hard to explain.

The idea is this:  the Beleths Drum has an electronic component that actually physically taps/knocks. You can feel in it your fingers if you hold it. If you use the screw-holes at the edges of the little circuit-board and screw it to a metal case, the percussion of the "knocking-thingy" is strong enough to make the device obnoxiously loud. Too loud.
In the 4000T chassis, I tried this, but it's like the inside of a Guitar's housing which a knocking-thingy on the inside, tapping away on the wood and due the workings of the housing, the sound is amplified (that is why Guitars are shaped as they are). It is LOUD.
So, I tried the foam-method like in the A3000D. Better, but still too loud.
I then resorted to letting it dangle in mid-air inside the case (roughly under the power-supply) so that the Beleths Drum does not come in contact with the case. This made the HDD sound audible enough to enjoy it, but does not drive me nuts.

**Reply by croissantking** [reply]
Thanks for the write-up - all very useful to know.
